cli/inputhandler/arghandler.cpp

```diff
@@ -9,9 +9,9 @@ namespace internal
 // Generic template function to check for user input value.
 // Expects input to be an unsigned long int.
 template<typename T>
-bool isUserInputValid(std::string const arg)
+bool isUserInputValid(std::string const &arg)
 {
-    if ((arg.size() == 0) || (arg.size() > std::numeric_limits<uint64_t>::digits)) return false;
+    if ((arg.empty()) || (arg.size() > std::numeric_limits<uint64_t>::digits)) return false;
```

Comment:
`std::numeric_limits<uint64_t>::digits` is already a variable inside a namespace. I don't see much benefits of storing it in another variable, especially because it is only being used once and doesn't really reduce readability. 
I would leave it as it is for now, however if you have a different opinion, I could change it, let me know